sources/Win32/Win32/DirectX/d3d12/d3dcommon/ID3DDestructionNotifier.gen.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+// Ported from d3dcommon.h in microsoft/DirectX-Headers tag v1.614.0
+// Original source is Copyright © Microsoft. Licensed under the MIT license
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Silk.NET.Windows;
+using static Silk.NET.Windows.IID;
+#pragma warning disable CS1589, CS0419, CA1416, CS0618
+namespace Silk.NET.DirectX;
+
+[Guid("A06EB39A-50DA-425B-8C31-4EECD6C270F3")]
+[NativeTypeName("struct ID3DDestructionNotifier : IUnknown")]
+[NativeInheritance("IUnknown")]
+public unsafe partial struct ID3DDestructionNotifier
+    : ID3DDestructionNotifier.Interface,
+        INativeGuid
+{
+    static Guid* INativeGuid.NativeGuid =>
+        (Guid*)Unsafe.AsPointer(ref Unsafe.AsRef(in IID_ID3DDestructionNotifier));
+    public void*** lpVtbl;
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    [VtblIndex(0)]
+    public HRESULT QueryInterface([NativeTypeName("const IID &")] Guid* riid, void** ppvObject)
+    {
+        return ((delegate* unmanaged<ID3DDestructionNotifier, Guid*, void**, int>)((*lpVtbl)[0]))(
+            this,
+            riid,
+            ppvObject
+        );
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    [VtblIndex(1)]
+    [return: NativeTypeName("ULONG")]
+    public uint AddRef()
+    {
+        return ((delegate* unmanaged<ID3DDestructionNotifier, uint>)((*lpVtbl)[1]))(this);
```

Comment:
Do we have any Thiscall methods anywhere in the full bindings? I believe there may be an ABI difference between `struct { void*** }` and `struct { void** }*`

Reply:
I don't believe so but shouldn't those two be functionally the same? ptr->ptr->ptr

Reply:
Dealt with this, ensuring whenever we are making calls o native we are doing Native* or Native** so abi should be good to go